src/GraphQL.php

```diff
@@ -175,6 +175,37 @@ public function addType($class, string $name = null): void
 
     public function type(string $name, bool $fresh = false): Type
     {
+        $modifiers = [];
+
+        while (true) {
+            if (preg_match('/^(.+)!$/', $name, $matches)) {
+                $name = $matches[1];
+                array_unshift($modifiers, 'nonNull');
+            } elseif (preg_match('/^\[(.+)\]$/', $name, $matches)) {
```

Comment:
Since you didn't start a character class, no need to escape the closing one
```suggestion
            } elseif (preg_match('/^\[(.+)]$/', $name, $matches)) {
```

Reply:
Already force-pushed the fix myself :p